jvm/src/main/java/io/glutenproject/memory/alloc/NativeMemoryAllocators.java

```diff
@@ -37,68 +33,33 @@ public static NativeMemoryAllocator contextInstance() {
       return globalInstance();
     }
 
-    final String id = NativeMemoryAllocatorManager.class.toString();
+    final String id = TaskMemoryResourceManager.class.toString();
```

Comment:
Can we add a interface `NativeMemoryAllocatorManager` then let `VeloxMemoryAllocatorManager` and `CHMemoryAllocatorManager` implement it? The interface can be empty if no sharing stuffs.

Then we can rewrite the line as 
```suggestion
    final String id = NativeMemoryAllocatorManager.class.toString();
```

Reply:
`TaskMemoryResourceManager` is a trait, we can change this trait to `NativeMemoryAllocatorManager`?

Reply:
I thought to let `TaskMemoryResourceManager` extends `NativeMemoryAllocatorManager`, then let `VeloxMemoryAllocatorManager`, `CHMemoryAllocatorManager` implement `TaskMemoryResourceManager`

By doing this the design becomes more reasonable because there was also `ArrowBufferAllocatorManager` implementing `TaskMemoryResourceManager`

Reply:
Fix